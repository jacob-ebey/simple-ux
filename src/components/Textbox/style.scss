@import '../variables';

.textbox {
  display: flex;
  flex-direction: column;
  padding: $content-padding / 2 0;

  &-inline {
    flex-direction: row;
    align-items: center;

    &>label {
      margin-right: $content-padding / 2;
    }
  }

  &>input, &>textarea {
    min-width: 100px;
    border: $border-width solid $color-dark;
    transition: all 0.30s ease-in-out;

    &:focus {
      border: $border-width solid $color-lessdark;
      box-shadow: 0 0 5px $color-primary;
    }
  }

  &>input {
    height: $input-height;
  }

  &>textarea {
    resize: vertical;
  }
}