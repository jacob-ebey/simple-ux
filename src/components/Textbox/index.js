import { h } from 'preact' // eslint-disable-line no-unused-vars

import { merge, mod } from '../tools'

import './style.scss'

let textboxCount = 0

const Textbox = ({ multiline, label, inline, id, class: className, ...rest }) => {
  const inputId = id || `textbox_${textboxCount++}`

  const Input = multiline ? 'textarea' : 'input'

  return (
    <div class={merge('textbox', mod('textbox', inline && 'inline'), className)}>
      { label && <label for={inputId}>{label}</label> }

      <Input
        {...rest}
        id={inputId}
      />
    </div>
  )
}

export default Textbox
