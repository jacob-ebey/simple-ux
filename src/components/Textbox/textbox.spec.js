/* eslint-env jest */
import { h } from 'preact' // eslint-disable-line no-unused-vars
import { shallow } from 'preact-render-spy'

import Textbox from './'

describe('Textbox', () => {
  describe('- snapshots -', () => {
    test('can render', () => {
      const wrapper = shallow(<Textbox />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add custom class', () => {
      const wrapper = shallow(<Textbox class='custom' />)
      expect(wrapper).toMatchSnapshot()
    })

    test('can render label', () => {
      const wrapper = shallow(<Textbox label='Test Label' />)
      expect(wrapper).toMatchSnapshot()
    })

    test('can render inline', () => {
      const wrapper = shallow(<Textbox inline />)
      expect(wrapper).toMatchSnapshot()
    })

    test('can render multiline', () => {
      const wrapper = shallow(<Textbox multiline />)
      expect(wrapper).toMatchSnapshot()
    })

    test('uses provided id', () => {
      const wrapper = shallow(<Textbox id='testid' />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
