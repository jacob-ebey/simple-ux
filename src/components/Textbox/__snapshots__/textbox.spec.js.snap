// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox - snapshots - can add custom class 1`] = `
preact-render-spy (1 nodes)
-------
<div class="textbox custom">
  <input id="textbox_1" />
</div>

`;

exports[`Textbox - snapshots - can render 1`] = `
preact-render-spy (1 nodes)
-------
<div class="textbox">
  <input id="textbox_0" />
</div>

`;

exports[`Textbox - snapshots - can render inline 1`] = `
preact-render-spy (1 nodes)
-------
<div class="textbox textbox-inline">
  <input id="textbox_3" />
</div>

`;

exports[`Textbox - snapshots - can render label 1`] = `
preact-render-spy (1 nodes)
-------
<div class="textbox">
  <label for="textbox_2">Test Label</label>
  <input id="textbox_2" />
</div>

`;

exports[`Textbox - snapshots - can render multiline 1`] = `
preact-render-spy (1 nodes)
-------
<div class="textbox">
  <textarea id="textbox_4"></textarea>
</div>

`;

exports[`Textbox - snapshots - uses provided id 1`] = `
preact-render-spy (1 nodes)
-------
<div class="textbox">
  <input id="testid" />
</div>

`;
